tests/clv/models/test_beta_geo.py

```diff
@@ -213,6 +214,76 @@ def test_numerically_stable_logp(
             decimal=5,
         )
 
+    @pytest.mark.parametrize("fit_type", ("map", "mcmc"))
+    def test_posterior_distributions(self, fit_type) -> None:
+        rng = np.random.default_rng(42)
+        dim_T = 2357
+
+        if fit_type == "map":
+            map_idata = self.model.idata.copy()
+            map_idata.posterior = map_idata.posterior.isel(
+                chain=slice(None, 1), draw=slice(None, 1)
+            )
+            model = self.model._build_with_idata(map_idata)
+            # We expect 1000 draws to be sampled with MAP
+            expected_shape = (1, 1000)
+            expected_pop_dims = (1, 1000, dim_T, 2)
+        else:
+            model = self.model
+            expected_shape = (self.chains, self.draws)
+            expected_pop_dims = (self.chains, self.draws, dim_T, 2)
+
+        data = model.data
+        customer_dropout = model.distribution_new_customer_dropout(
+            data, random_seed=rng
+        )
+        customer_purchase_rate = model.distribution_new_customer_purchase_rate(
+            data, random_seed=rng
+        )
+        customer_rec_freq = model.distribution_new_customer_recency_frequency(
+            data, random_seed=rng
+        )
+        customer_rec = customer_rec_freq.sel(obs_var="recency")
+        customer_freq = customer_rec_freq.sel(obs_var="frequency")
+
+        assert customer_dropout.shape == expected_shape
+        assert customer_purchase_rate.shape == expected_shape
+        assert customer_rec_freq.shape == expected_pop_dims
+
+        lam_mean = self.r_true / self.alpha_true
+        lam_std = np.sqrt(self.r_true) / self.alpha_true
+        mu_mean = self.s_true / self.beta_true
+        mu_std = np.sqrt(self.s_true) / self.beta_true
+        ref_rec, ref_freq = pm.draw(
+            BGNBD.dist(
+                r=self.r_true,
+                alpha=self.alpha_true,
+                s=self.s_true,
+                beta=self.beta_true,
+                T=self.T,
+            ),
+            random_seed=rng,
+        ).T
+
+        np.testing.assert_allclose(
+            customer_purchase_rate.mean(),
+            lam_mean,
+            rtol=0.5,
+        )
+        np.testing.assert_allclose(
+            customer_purchase_rate.std(),
+            lam_std,
+            rtol=0.5,
+        )
+        np.testing.assert_allclose(customer_dropout.mean(), mu_mean, rtol=0.5)
+        np.testing.assert_allclose(customer_dropout.std(), mu_std, rtol=0.5)
+
+        np.testing.assert_allclose(customer_rec.mean(), ref_rec.mean(), rtol=0.5)
+        np.testing.assert_allclose(customer_rec.std(), ref_rec.std(), rtol=0.5)
+
+        np.testing.assert_allclose(customer_freq.mean(), ref_freq.mean(), rtol=0.5)
+        np.testing.assert_allclose(customer_freq.std(), ref_freq.std(), rtol=0.5)
```

Comment:
The new distribution block will need to be added to `_distribution_new_customers`. Here's an [example](https://github.com/pymc-labs/pymc-marketing/blob/main/pymc_marketing/clv/models/pareto_nbd.py#L947) of what would look like.

`_distribution_new_customers` will also need to be overridden for `ModifiedBetaGeoModel` if a block for that model isn't added before the next release.